xblock/mixins.py

```diff
@@ -522,7 +522,9 @@ def _add_field(self, node, field_name, field):
             elem.text = value
             node.insert(0, elem)
         else:
-            node.set(field_name, value)
+            # setting the None in the field_name raises TypeError to avoid this
```

Comment:
when value of attribute is None e.g ```giturl``` which is ```setting``` field and if its not set will raise ```TypeError``` because of its ```None``` value